### ISSUE TITLE
Feature Request: Ability to set nested values using `make` method

### ISSUE THREAD

**Issue by ishanvyas22** [vierge-noire/cakephp-fixture-factories#207]
## Problem

Currently it is not possible to set values that are nested.

```php
class ExampleFactory extends BaseFactory
{
    ...

    protected function setDefaultTemplate(): void
    {
        $this->setDefaultData(function (Generator $faker) {
            return [
                'name' => $faker->name,
                'data' => [
                    'mobile' => 123456,
                    'email' => $faker->email,
                ],
                'created' => FrozenTime::now(),
                'modified' => FrozenTime::now(),
            ];
        });
    }

    ...
}
```

For above factory, this works: 
```php
ExampleFactory::make(['name' => 'John Doe'])->persist();
```

this doesn't: 
```php
ExampleFactory::make(['data' => ['email' => 'foo@example.org']])->persist();
```

Above will overwrite the whole `data` array values instead of just updating the `data.email` field.

## Proposal

It would be a nice to allow developers to set nested values as well. 

#### Option 1

Simply provide array:
```php
ExampleFactory::make(['data' => ['email' => 'foo@example.org']])->persist();
```

#### Option 2

Using dot notation:
```php
ExampleFactory::make(['data.email' => 'foo@example.org'])->persist();
```

Whichever options works best.

**Comment by pabloelcolombiano** [comment]
Thanks for the proposal @ishanvyas22 !
The dot notation is indeed a nice way to specify that a nested value should be inserted.

@pakacuda is this good for you?

**Comment by pabloelcolombiano** [comment]
Merged with v3.9
